src/io/sb3/fromSb3.ts

```diff
@@ -85,93 +89,104 @@ function getBlockScript(blocks: { [key: string]: sb3.Block }) {
-                      };
-                    case "%b":
-                      return {
-                        type: "boolean",
-                        name: argNames.splice(0, 1)[0],
-                        defaultValue: argDefaults.splice(0, 1)[0] === "true"
-                      };
-                    default:
-                      return {
-                        type: "label",
-                        name: part
-                      };
                   }
-                });
-
-                addInput("PROCCODE", { type: "string", value: mutation.proccode });
-                addInput("ARGUMENTS", { type: "customBlockArguments", value: args });
-                addInput("WARP", { type: "boolean", value: mutation.warp === "true" });
-              } else {
-                // In most cases, just copy the shadow block's fields and inputs
-                // into its parent
-                result = {
-                  ...result,
-                  ...translateInputs(blocks[value].inputs),
-                  ...translateFields(blocks[value].fields, blocks[value].opcode)
+                  const asNum = parseFloat(value);
+                  if (!isNaN(asNum)) {
+                    return asNum;
+                  }
+                  return value;
                 };
+
+                switch (part) {
+                  case "%s":
+                  case "%n":
+                    return {
+                      type: "numberOrString",
+                      name: argNames.splice(0, 1)[0],
+                      defaultValue: optionalToNumber(argDefaults.splice(0, 1)[0])
+                    };
+                  case "%b":
+                    return {
+                      type: "boolean",
+                      name: argNames.splice(0, 1)[0],
+                      defaultValue: argDefaults.splice(0, 1)[0] === "true"
+                    };
+                  default:
+                    return {
+                      type: "label",
+                      name: part
+                    };
+                }
+              });
+
+              addInput("PROCCODE", { type: "string", value: mutation.proccode });
+              addInput("ARGUMENTS", { type: "customBlockArguments", value: args });
+              addInput("WARP", { type: "boolean", value: mutation.warp === "true" });
+            } else {
+              // In most cases, just copy the shadow block's fields and inputs
+              // into its parent
+              result = {
+                ...result,
+                ...translateInputs(blocks[value].inputs),
+                ...translateFields(blocks[value].fields, blocks[value].opcode)
+              };
+            }
+          } else {
+            let isBlocks;
+            if (BlockBase.isKnownBlock(block.opcode)) {
+              const { type } = BlockBase.getDefaultInput(block.opcode, inputName) || {};
+              if (type === "blocks") {
+                isBlocks = true;
               }
```

Comment:
```suggestion
              const defaultInput = BlockBase.getDefaultInput(block.opcode, inputName);
              if (defaultInput && defaultInput.type === "blocks") {
                isBlocks = true;
              }
```
I think this reads a bit clearer